### PR TITLE
Sentinel 2 NDVI script for remote data module

### PR DESCRIPTION
**Pull request by ayushprd**
## Description
This PR uses satellitetool's functions to calculate NDVI from Sentinel 2 data. 

## Motivation and Context

These scripts will be used to create an automated workflow for the remote data module.
To test the PR please install these python dependencies [requirements.txt](https://github.com/ayushprd/PEcAn-GEE/blob/master/requirements.txt) These will later be added to the PEcAn dependencies. 

## Review Time Estimate

- [ ] Immediately
- [ ] Within one week
- [x] When possible
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) 
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: modules/data.remote/inst/satellitetools/gee.py [hunk 1]
```diff
@@ -0,0 +1,766 @@
...
+    s2_data_feature_collection = feature_collection.map(
+        ee_get_s2_data_feature).getInfo()
+
+    s2_data = s2_feature_collection_to_dataframes(s2_data_feature_collection)
+
+    for a in AOIs:
+        name = a.name
+        a.data = s2_data[name]
+
+# Old, unused function, remove at some point.
```
Comment (istfer) [inline modules/data.remote/inst/satellitetools/gee.py:450]: Is it safe to remove this already?

Comment (ayushprd) [reply modules/data.remote/inst/satellitetools/gee.py]: Yes, removed.

File: modules/data.remote/inst/s2ndvi.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    
+    geofile (str) -- path to the file containing the coordinates of AOI, currently tested with geojson. 
+    
+    outdir (str) -- path to the directory where the output file is stored. If specified directory does not exists, it is created.
+  
+    start (str) -- starting date of the data request in the form YYYY-MM-DD
+    
+    end (str) -- ending date of the data request in the form YYYY-MM-DD
+    
+    qi_threshold (float) -- From satellitetools: Threshold value to filter images based on used qi filter. qi filter holds labels of classes whose percentages within the AOI is summed. If the sum is 				     larger then the qi_threhold, data will not be retrieved for that date/image. The default is 1, meaning all data is retrieved
```
Comment (istfer) [inline modules/data.remote/inst/s2ndvi.py:33]:
```suggestion
    qi_threshold (float) -- From satellitetools: Threshold value to filter images based on used qi filter. qi filter holds labels of classes whose percentages within the AOI is summed. If the sum is larger then the qi_threshold, data will not be retrieved for that date/image. The default is 1, meaning all data is retrieved
```

File: modules/data.remote/inst/s2ndvi.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    # convert dataframe to an xarray dataset, used later for converting to netCDF
+    gee.s2_data_to_xarray(area, request)
+
+    # calculate NDVI for the selected area
+    area.data = gee.compute_ndvi(area.data)
+
+    timeseries = {}
+    timeseries_variable = ["ndvi"]
+
+    # is specified output directory does not exist, create it.
```
Comment (istfer) [inline modules/data.remote/inst/s2ndvi.py:66]:
```suggestion
    # if specified output directory does not exist, create it.
```

File: modules/data.remote/inst/satellitetools/gee.py [hunk 4]
```diff
@@ -0,0 +1,716 @@
...
+
+def xr_dataset_to_timeseries(xr_dataset, variables):
+    """Compute timeseries dataframe from xr dataset.
+
+    Parameters
+    ----------
+    xr_dataset : xarray dataset
+
+    variables : list
+        list of varbiale names as string.
```
Comment (istfer) [inline modules/data.remote/inst/satellitetools/gee.py:685]:
```suggestion
        list of variable names as string.
```

File: modules/data.remote/inst/s2ndvi.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+"""
+Module to calculate NDVI from sentinel 2 data, based on Satellitetools by Olli Nevalainen.
+This will be used to create automated workflows for the remote data module.
+
+Warning: Point coordinates as input has currently not been implemented.
+
+Author: Ayush Prasad
+"""
+
+from satellitetools import gee
```
Comment (istfer) [inline modules/data.remote/inst/s2ndvi.py:13]: I'm not familiar with python, but just to confirm, does this also work without having satellitetools as a library?

Comment (ayushprd) [reply modules/data.remote/inst/s2ndvi.py:13]: satellitetools is just the name of the folder which contains the original script `gee.py`, this script is required.

Comment (istfer) [reply modules/data.remote/inst/s2ndvi.py:13]: Yup that much I followed, I wasn't just aware if one can simply import a function from a folder

**Comment by istfer** [comment]
I know this is not a complete work and will be built upon, but would it still be nice to include a small vignette to accompany this PR? Nothing too fancy, just the main steps to be able to make the Reykjavik example work, where you can also list the python dependencies?

**Comment by ayushprd** [comment]
Yes, @istfer I'll definitely do that, I didn't add that in the docstring right now because, the arguments to the function may be change in the very near future.

File: modules/data.remote/inst/s2ndvi.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    Returns
+    -------
+    Nothing:
+            output netCDF is saved in the specified directory.
+            
+    Python dependencies required: earthengine-api, geopandas, pandas, netCDF4, xarray
+    
+    To test this function please run the following code, testfile is included.
+    
+    s2_ndvi(geofile="./test.geojson", outdir="./out/", start="2019-01-01", end="2019-12-31", qi_threshold=1)
```
Comment (istfer) [inline modules/data.remote/inst/s2ndvi.py:45]: could you move `test.geojson` into `satellitetools` folder and update this, not a big deal but that feels like a more appropriate place

Comment (ayushprd) [reply modules/data.remote/inst/s2ndvi.py:45]: Sure.

**Comment by robkooper** [comment]
@dlebauer I think this is a good case to squash before merge, since some files were added and removed later. If we don't squash when we do the merge, this will result in multiple commits that add/remove files and will grow the repository without really need be.

**Pull request merged**